### ISSUE TITLE
Background push not working  ios

### ISSUE THREAD

**Issue by gopaldevra** [firebase/quickstart-ios#137]
Hi
Push notifications are working in foreground but not in the background.
Actually push receives but in the notification area it doesn't show.
Please let me know what could be the issue.
Thanks

**Comment by evgenii22** [comment]
+1

**Comment by Muneefm** [comment]
@gopaldevra  same happening for me too. tried everything.  have you found a solution yet?

**Comment by concentrateOne** [comment]
+10086

**Comment by evgenii22** [comment]
Fuck Firebase,
Fuck Google.

**Comment by ianbarber** [comment]
Sorry for the delay on a response. I am not sure I understand the "push receives but in the notification area it doesn't show". 

"data" type messages are sent to the device - are you sending a message with a "notification" key? That should be sent via APNs and appear as a regular device notification. 

Any details you have of the type of message that works or doesn't would be helpful.

**Comment by MeteC** [comment]
I'm experiencing something similar (I think), in that I get pushes in the foreground but nothing in the background, for apps that I update from a previous version (that is, a version using GCM not FCM). Nothing comes when the app is backgrounded, ever. BUT when I uninstall the app completely, then reinstall it with the FCM-enabled version of my app, it works just fine.

So it's like it's working, but not for app upgrade, which is painful... Any clues?

I've tried with messages using "data", "notification", and both "data" + "notification" elements. It just doesn't seem to be going through the APNs - and I've also tried manually setting the FIRInstance APNS token despite swizzling enabled. Nothing.

**Comment by revolter** [comment]
I noticed that exact thing that @MeteC described, worked only after a fresh install. Though single device push notifications worked correctly in either case.

**Comment by kroikie** [comment]
@gopaldevra have you tried setting the notification priority to HIGH?

**Comment by monish152** [comment]
I am facing same issue in ios9

**Comment by sundarsan** [comment]
we are sending the both Data & Notification payload with "Content-available = true" flag and Priortity : High . i am also facing the similar kind of issue . i receive the payload fine when the app is running in foreground. but as soon as app goes to background i am not receiving anything.

Am i doing anything wrong ?

**Comment by morganchen12** [comment]
Hey all,

Normally FCM data messages will be sent directly through FCM (bypassing APNs) while the app is foregrounded. If you're running into this issue, double check your APNs setup first. You can test your APNs setup first by sending notifications directly through APNs.

If you're still having trouble, file a new issue with your code and the steps you took to set up FCM.
